envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -210,18 +211,61 @@ class EnvoyClustersFactory(
         return clustersForGroup
     }
 
-    private fun createClusterForGroup(dependencySettings: DependencySettings, cluster: Cluster?): Cluster? {
+    private fun createClusterForGroup(
+        dependencySettings: DependencySettings,
+        cluster: Cluster?,
+        defaultDependencySettings: DependencySettings? = null
+    ): Cluster? {
         return cluster?.let {
             val idleTimeoutPolicy =
-                dependencySettings.timeoutPolicy.connectionIdleTimeout ?: cluster.commonHttpProtocolOptions.idleTimeout
+                dependencySettings.timeoutPolicy.connectionIdleTimeout
+                    ?: defaultDependencySettings?.timeoutPolicy?.connectionIdleTimeout
+                    ?: cluster.commonHttpProtocolOptions.idleTimeout
             Cluster.newBuilder(cluster)
+                .setCircuitBreakers(createCircuitBreakers(dependencySettings))
                 .setCommonHttpProtocolOptions(
                     HttpProtocolOptions.newBuilder().setIdleTimeout(idleTimeoutPolicy)
                 ).build()
         }
     }
 
-    private fun shouldAddDynamicForwardProxyCluster(group: Group) =
+    private fun createCircuitBreakers(
+        dependencySettings: DependencySettings,
+        defaultDependencySettings: DependencySettings? = null
+    ): CircuitBreakers {
+        val defaultThreshold = dependencySettings.circuitBreakers.defaultThreshold
+            ?: defaultDependencySettings?.circuitBreakers?.defaultThreshold
+        val highThreshold = dependencySettings.circuitBreakers.highThreshold
+            ?: defaultDependencySettings?.circuitBreakers?.highThreshold
+        val thresholds = listOf(
+            defaultThreshold?.toThreshold(RoutingPriority.DEFAULT),
+            highThreshold?.toThreshold(RoutingPriority.HIGH)
+        ).filterNotNull()
```

Comment:
this could be simplified 
```suggestion
        val thresholds = listOfNotNull(
            defaultThreshold?.toThreshold(RoutingPriority.DEFAULT),
            highThreshold?.toThreshold(RoutingPriority.HIGH)
        )
```

Reply:
changed